app/app.go

```diff
@@ -367,31 +373,88 @@ func New(
-		module.NewVersionedModule(blobstream.NewAppModule(appCodec, app.BlobstreamKeeper), v1.Version, v2.Version),
-		module.NewVersionedModule(upgrade.NewAppModule(app.UpgradeKeeper), v2.Version, v2.Version),
-	)
+	app.mm, err = module.NewManager([]module.VersionedModule{
+		{
+			Module:      genutil.NewAppModule(app.AccountKeeper, app.StakingKeeper, app.BaseApp.DeliverTx, encodingConfig.TxConfig),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      auth.NewAppModule(appCodec, app.AccountKeeper, nil),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      vesting.NewAppModule(app.AccountKeeper, app.BankKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      bank.NewAppModule(appCodec, app.BankKeeper, app.AccountKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      capability.NewAppModule(appCodec, *app.CapabilityKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      feegrantmodule.NewAppModule(appCodec, app.AccountKeeper, app.BankKeeper, app.FeeGrantKeeper, app.interfaceRegistry),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      crisis.NewAppModule(&app.CrisisKeeper, skipGenesisInvariants),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      gov.NewAppModule(appCodec, app.GovKeeper, app.AccountKeeper, app.BankKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      mint.NewAppModule(appCodec, app.MintKeeper, app.AccountKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      slashing.NewAppModule(appCodec, app.SlashingKeeper, app.AccountKeeper, app.BankKeeper, app.StakingKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      distr.NewAppModule(appCodec, app.DistrKeeper, app.AccountKeeper, app.BankKeeper, app.StakingKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      staking.NewAppModule(appCodec, app.StakingKeeper, app.AccountKeeper, app.BankKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      evidence.NewAppModule(app.EvidenceKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      authzmodule.NewAppModule(appCodec, app.AuthzKeeper, app.AccountKeeper, app.BankKeeper, app.interfaceRegistry),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      ibc.NewAppModule(app.IBCKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      params.NewAppModule(app.ParamsKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      transfer.NewAppModule(app.TransferKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      blob.NewAppModule(appCodec, app.BlobKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      blobstream.NewAppModule(appCodec, app.BlobstreamKeeper),
+			FromVersion: v1, ToVersion: v2,
+		},
+		{
+			Module:      upgrade.NewAppModule(app.UpgradeKeeper),
+			FromVersion: v2, ToVersion: v2,
+		},
+	})
```

Comment:
I made this change predominantly for readability reasons but happy to revert if people think this does not help

Reply:
Looks fine to me